### ISSUE TITLE
Mac: Starting another backend

### ISSUE THREAD

**Issue by jermy-c** [mustang-im/mustang#317]
**Reproduction**

1. Open Mustang
2. Close Mustang, don't quit the app
3. Open Mustang again

**Actual Result**

1. Displays an error dialog
2. Click "OK"
3. The app starts again

**Expected Result**

1. Displays the app

<img width="372" alt="圖片" src="https://github.com/user-attachments/assets/403659ec-d29d-4357-b215-f9bbe54da594" />

**Comment by NeilRashbrook** [comment]
I created commit 47e4860 which I think might address this to some extent but I'd have to dig out my ancient Mac and try and install Node on it to fully test it so I don't suppose you could try it out?

**Comment by benbucksch** [comment]
@jermy-c Could you please try this proposed fix?

**Comment by benbucksch** [comment]
Branch `neil/restart-backend`

This is stopping the backend when the main window closes, and restarts it when the window opens. I reviewed the code, and it looks right to me - if we follow this logic.

However, that fix means that once the main window is closed, we will not do any mail check anymore (which we currently don't do anyways), and we will not be able to do that anymore.

Additionally, if we were to open a second main window, and close it, the backend would die and leave the first main window stranded. Again, that's not a case that we have right now. But it's something that might potentially come up in the future.

**Comment by benbucksch** [comment]
@NeilRashbrook Can you make that into a PR? The code looks good to me and is a step forward.

**Comment by jermy-c** [comment]
It works I don't see the dialog but IMAP has a problem after closing and opening 2-3 times without quitting.

```
index-uad3MjUS.js:37969 LoginError: Check your login, username, and password.
Too many simultaneous connections. (Failure)
    at IMAPAccount.connection (index-uad3MjUS.js:66563:35)
    at async IMAPAccount.login (index-uad3MjUS.js:66496:5)
    at async loadFolder (index-uad3MjUS.js:386869:9)
```

**Comment by benbucksch** [comment]
a) We need to stop IMAP connections and all other persistent connections. This should likely happen from the frontend in an unload handler.
b) We need to drop all objects in the backend that are no longer needed. Specifically, the callbacks needs to be removed. These are the reason for bug #61.

In other words, we need to shut down from 2 sides: The frontend needs to power down everything that it can think of, and the backend needs to drop everything that remains, because we're likely to have missed something, now or in the far future, and we don't want to leak it.

**Comment by benbucksch** [comment]
@jermy-c This might be fixed now with the mailto: changes. Can you please test whether this bug still exists?

**Comment by jermy-c** [comment]
This is the bug mentioned in https://github.com/mustang-im/mustang/issues/317#issuecomment-3049489393 or the comment above.

<img width="372" height="388" alt="Image" src="https://github.com/user-attachments/assets/626e091d-f7a0-4c4d-8391-b4b77bf8fd99" />

**Comment by jermy-c** [comment]
> @jermy-c This is a different error, now about TLS. Is the original bug here fixed? 

Yes, the original bug is fixed.

> FWIW, you might be able to reproduce this without waiting by explicitly putting the device to sleep, right? Or maybe not, if this is about a timeout from the server side who didn't hear from us anymore. But the timeout is likely much shorter?

The above was in dev mode with `yarn dev`. I did a build(`yarn build:mac`) today and ran it and I didn't get the error so far.

**Comment by benbucksch** [comment]
> Yes, the original bug is fixed.

That's great. The original was the difficult one.

Could you please file a new bug with a reproduction for the TLS timeout, please?

> build(yarn build:mac) today and ran it and I didn't get the error so far.

I don't think this TLS bug is related to the build at all, so that's coincidental.
